tests/codegen/waiter-tests/src/test/kotlin/com/test/WaiterTest.kt

```diff
@@ -1014,6 +1014,112 @@ class WaiterTest {
         },
     )
 
+    // function abs
+    @Test fun testAbsFunctionShortEquals() = successTest(
+        WaitersTestClient::waitUntilAbsFunctionShortEquals,
+        GetEntityResponse { primitives = EntityPrimitives { short = -1 } },
+        GetEntityResponse { primitives = EntityPrimitives { short = 1 } },
+        GetEntityResponse { primitives = EntityPrimitives { short = 2 } },
+    )
```

Comment:
**Suggestion**: Our test code and our test model are too big. I suggest looking into ways to split up the code/model by JMESPath feature. For instance, we could have separate test classes for `abs`, `ceil`, `floor`, primitive equality, comparators, list indexing, etc.

Similarly, we could break up the model into smaller files. This would be a lot trickier because right now all these waiters exist on a single operation in a single service. We could consider separate services/operations for each group. We could also consider defining multiple files that just contain groups of waiter definitions and then stitch them together into **waiter-operations.smithy** during the build process.